pkg/security/auth.go

```diff
@@ -141,8 +149,19 @@ func UserAuthCertHook(insecureMode bool, tlsState *tls.ConnectionState) (UserAut
 			return errors.Errorf("using tenant client certificate as user certificate is not allowed")
 		}
 
-		// The client certificate user must match the requested user.
-		if !Contains(certUsers, systemIdentity.Normalized()) {
+		// If the certificate is a tenant scoped client certificate, we should enforce that the tenant ID
+		// and client name matches with the certificate. Otherwise, it is sufficient to just check that the
+		// client name matches the certificate.
+		// TODO(rima): Should we enforce always using tenant scoped client cert for non-system tenants?
```

Comment:
I'd argue we should not allow the use of client certificates without the `crdb://tenant/<id>/user/<username>` SAN in a multi-tenant system.

The system tenants, such as node users, should use `crdb://node/<node id>` certificates to differentiate them, and we could add logic to prevent them from having username SANs in them as well.